src/ApiService/ApiService/onefuzzlib/NotificationOperations.cs

```diff
@@ -0,0 +1,154 @@
+            }
+
+            if (report == null)
+            {
+                continue;
+            }
+
+            if (notification.Config.AdoTemplate != null)
+            {
+                NotifyAdo(notification.Config.AdoTemplate, container, filename, report, failTaskOnTransientError);
+            }
+
+            if (notification.Config.GithubIssuesTemplate != null)
+            {
+                GithubIssue(notification.Config.GithubIssuesTemplate, container, filename, report);
+            }
+        }
+
+        await foreach (var (task, containers) in GetQueueTasks())
+        {
+            if (containers.Contains(container.ContainerName))
+            {
+                _log.Info($"queuing input {container.ContainerName} {filename} {task.TaskId}");
+                var url = _containers.GetFileSasUrl(container, filename, StorageType.Corpus, read: true, delete: true);
+                await _queue.SendMessage(task.TaskId.ToString(), System.Text.Encoding.UTF8.GetBytes(url.ToString()), StorageType.Corpus);
+            }
+        }
+
+        if (report == null)
+        {
+            await _events.SendEvent(new EventFileAdded(container, filename));
+        }
+        else if (report.Report != null)
+        {
+            var reportTask = await _taskOperations.GetByTaskId(report.Report.TaskId);
+            TaskConfig? taskConfig = null;
+            if (reportTask != null)
+            {
+                taskConfig = reportTask.Config;
+            }
+            var crashReportedEvent = new EventCrashReported(report.Report, container, filename, taskConfig);
+            await _events.SendEvent(crashReportedEvent);
+        }
+        else if (report.RegressionReport != null)
+        {
+            var reportTask = await GetRegressionReportTask(report.RegressionReport);
+            TaskConfig? taskConfig = null;
+            if (reportTask != null)
+            {
+                taskConfig = reportTask.Config;
+            }
+            var regressionEvent = new EventRegressionReported(report.RegressionReport, container, filename, taskConfig);
+        }
+    }
+
+    public IAsyncEnumerable<Notification> GetNotifications(Container container)
+    {
+        return QueryAsync(filter: $"container eq '{container.ContainerName}'");
+    }
+
+    public IAsyncEnumerable<(Task, IEnumerable<string>)> GetQueueTasks()
+    {
+        // Nullability mismatch: We filter tuples where the containers are null
+        return _taskOperations.SearchStates(states: TaskStateHelper.Available())
+            .Select(task => (task, _taskOperations.GetInputContainerQueues(task.Config)))
+            .Where(taskTuple => taskTuple.Item2 != null)!;
+    }
+
+    private async Async.Task<Task?> GetRegressionReportTask(RegressionReport report)
+    {
+        if (report.CrashTestResult.CrashReport != null)
+        {
+            return await _taskOperations.GetByTaskId(report.CrashTestResult.CrashReport.TaskId);
```

Comment:
In the python code we also use job id as part of the filter. Was that a query optimization?

Reply:
yes it is faster to query with both PartitionKey and RowKey than just RowKey

Reply:
Changed it to use both job id and task id